plugins/state_history_plugin/state_history_plugin.cpp

```diff
@@ -351,6 +351,7 @@ struct state_history_plugin_impl : std::enable_shared_from_this<state_history_pl
 
```

Comment:
I am curious about the `if (!current_request || !current_request->max_messages_in_flight)` above. Why return when `max_messages_in_flight` is false?

Reply:
So why the second part not `current_request->max_messages_in_flight` but `!current_request->max_messages_in_flight`?

Reply:
Ah, look at line 342, it counts down.